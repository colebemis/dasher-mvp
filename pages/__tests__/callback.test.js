import React from 'react'
import CallbackPage from '../callback'
import { render, cleanup } from 'react-testing-library'
import Router from 'next/router'

jest.mock('../../lib/github')

it('renders without crashing', () => {
  render(<CallbackPage />)
  cleanup()
})

describe('getInitialProps', () => {
  it('sets cookies and redirects to /', async () => {
    const context = {
      res: {
        setHeader: jest.fn(),
        writeHead: jest.fn(),
        end: jest.fn(),
      },
      query: {
        code: 'TEST_CODE',
      },
    }

    await CallbackPage.getInitialProps(context)

    expect(context.res.setHeader.mock.calls).toMatchSnapshot()
    expect(context.res.writeHead.mock.calls).toMatchSnapshot()
  })

  it('redirects to /login if `res` is undefined', async () => {
    await CallbackPage.getInitialProps({})

    expect(Router.push).toHaveBeenCalledWith('/login')
  })
})
