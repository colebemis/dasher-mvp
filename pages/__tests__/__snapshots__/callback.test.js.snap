// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInitialProps sets cookies and redirects to / 1`] = `
Array [
  Array [
    "github_token",
    "TEST_GITHUB_TOKEN",
  ],
]
`;

exports[`getInitialProps sets cookies and redirects to / 2`] = `
Array [
  Array [
    302,
    Object {
      "Location": "/",
    },
  ],
]
`;
